### PR TITLE
Replace document.title with vue-meta

### LINKED ISSUE #910: Use vue-meta to manage pages titles
https://www.npmjs.com/package/vue-meta

This refactoring is needed for SSR. To don't refer to `document` explicitly in the router.

https://github.com/aeternity/superhero-ui/blob/3b07c6deb19a60eddcca173baad7365936a51d0f/src/router.js#L194

### PR DESCRIPTION
**Pull request by shapkarin**
closes #910

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -110,6 +110,14 @@ export default {
       await this.reloadUserData();
     },
   },
+  metaInfo() {
+    return {
+      // default title
+      title: 'Superhero.com',
```
Comment (davidyuk) [inline src/App.vue:116]: This is not working this way

File: src/App.vue [hunk 2]
```diff
@@ -110,6 +110,14 @@ export default {
       await this.reloadUserData();
     },
   },
+  metaInfo() {
+    return {
+      // default title
```
Comment (davidyuk) [inline src/App.vue:115]: Let's leave it for the documentation of `vue-meta`

File: src/App.vue [hunk 3]
```diff
@@ -110,6 +110,14 @@ export default {
...
+      title: 'Superhero.com',
+      // will be injected into page meta
+      titleTemplate: `${this.$route.meta.title} | Superhero.com`,
```
Comment (davidyuk) [inline src/App.vue:118]:
```suggestion
      titleTemplate: `${this.$route.meta.title} - Superhero.com`,
```
be consistent with the previous implementation, I don't want to change the title format now

**Review by davidyuk — COMMENTED** [review]
Why you didn't add meta for each page? See how it is powerful using this way: a41e1ce888e92919885e8c64980a0d0f1fb77ef8
Proposing to merge #915 instead

File: src/App.vue (hunk 1 above)
Comment (shapkarin) [reply src/App.vue:116]: It default fallback for the case when route don't have any meta title

File: src/App.vue (hunk 2 above)
Comment (shapkarin) [reply src/App.vue:115]: ok

**Comment by shapkarin** [comment]
> Why you didn't add meta for each page? See how it is powerful using this way: [a41e1ce](https://github.com/aeternity/superhero-ui/commit/a41e1ce888e92919885e8c64980a0d0f1fb77ef8)
> Proposing to merge #915 instead

Not sure why you decide to place the title to the component but OK 
My implementation does pretty the same thing – show titles. But they defined at routes as it was before.

**Comment by shapkarin** [comment]
@davidyuk you think that it makes easier to keep all about page in the page component itself, right?
@mradkov please, share your thoughts too maybe @davidyuk point is right and it's better to place titles to the each component.

File: src/App.vue (hunk 3 above)
Comment (shapkarin) [reply src/App.vue:118]: right

**Pull request closed without merging**